litellm/main.py

```diff
@@ -300,6 +300,182 @@ async def create(self, messages, model=None, **kwargs):
+    stream: Literal[False],
+    stream_options: Optional[dict] = None,
+    stop=None,
+    max_tokens: Optional[int] = None,
+    max_completion_tokens: Optional[int] = None,
+    modalities: Optional[List[ChatCompletionModality]] = None,
+    prediction: Optional[ChatCompletionPredictionContentParam] = None,
+    audio: Optional[ChatCompletionAudioParam] = None,
+    presence_penalty: Optional[float] = None,
+    frequency_penalty: Optional[float] = None,
+    logit_bias: Optional[dict] = None,
+    user: Optional[str] = None,
+    # openai v1.0+ new params
+    response_format: Optional[Union[dict, Type[BaseModel]]] = None,
+    seed: Optional[int] = None,
+    tools: Optional[List] = None,
+    tool_choice: Optional[str] = None,
+    parallel_tool_calls: Optional[bool] = None,
+    logprobs: Optional[bool] = None,
+    top_logprobs: Optional[int] = None,
+    deployment_id=None,
+    # set api_base, api_version, api_key
+    base_url: Optional[str] = None,
+    api_version: Optional[str] = None,
+    api_key: Optional[str] = None,
+    model_list: Optional[list] = None,  # pass in a list of api_base,keys, etc.
+    extra_headers: Optional[dict] = None,
+    # Optional liteLLM function params
+    **kwargs,
+) -> ModelResponse: ...
+
+
+@overload
+async def acompletion(
+    model: str,
+    *,
+    # Optional OpenAI params: see https://platform.openai.com/docs/api-reference/chat/create
+    messages: List = [],
+    functions: Optional[List] = None,
+    function_call: Optional[str] = None,
+    timeout: Optional[Union[float, int]] = None,
+    temperature: Optional[float] = None,
+    top_p: Optional[float] = None,
+    n: Optional[int] = None,
+    stream: Literal[True],
+    stream_options: Optional[dict] = None,
+    stop=None,
+    max_tokens: Optional[int] = None,
+    max_completion_tokens: Optional[int] = None,
+    modalities: Optional[List[ChatCompletionModality]] = None,
+    prediction: Optional[ChatCompletionPredictionContentParam] = None,
+    audio: Optional[ChatCompletionAudioParam] = None,
+    presence_penalty: Optional[float] = None,
+    frequency_penalty: Optional[float] = None,
+    logit_bias: Optional[dict] = None,
+    user: Optional[str] = None,
+    # openai v1.0+ new params
+    response_format: Optional[Union[dict, Type[BaseModel]]] = None,
+    seed: Optional[int] = None,
+    tools: Optional[List] = None,
+    tool_choice: Optional[str] = None,
+    parallel_tool_calls: Optional[bool] = None,
+    logprobs: Optional[bool] = None,
+    top_logprobs: Optional[int] = None,
+    deployment_id=None,
+    # set api_base, api_version, api_key
+    base_url: Optional[str] = None,
+    api_version: Optional[str] = None,
+    api_key: Optional[str] = None,
+    model_list: Optional[list] = None,  # pass in a list of api_base,keys, etc.
+    extra_headers: Optional[dict] = None,
+    # Optional liteLLM function params
+    **kwargs,
+) -> CustomStreamWrapper: ...
+
+
+@overload
+async def acompletion(
```

Comment:
there is a lot of duplicate code in this approach, this means that when OAI has a new param we'll need to edit it in 5-6 places. Do you have a better approach in mind with less code duplication @miraclebakelaser ?